### PR TITLE
Add pipeline-service-storage argocd app

### PR DESCRIPTION
**Pull request by Mo3m3n**
The pipeline-service-storage app will install both postgres and minio used only for development purpose

### CODE REVIEW

**Comment by Roming22** [comment]
There’s an issue during deployment:
```
+ kubectl apply -k /source/developer/openshift/work/credentials/manifests/compute/tekton-results
error: accumulating resources: accumulation err='accumulating resources from 'tekton-results-secret.yaml': evalsymlink failure on '/source/developer/openshift/work/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml' : lstat /source/developer/openshift/work/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml: no such file or directory': must build at directory: not a valid directory: evalsymlink failure on '/source/developer/openshift/work/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml' : lstat /source/developer/openshift/work/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml: no such file or directory
ry-run=client -o yaml
```

It looks like a kustomization.yaml is referencing something it shouldn’t, or dev_setup.sh is not creating a file that is needed.

**Comment by Mo3m3n** [comment]
> There’s an issue during deployment:
> 
> ```
> + kubectl apply -k /source/developer/openshift/work/credentials/manifests/compute/tekton-results
> error: accumulating resources: accumulation err='accumulating resources from 'tekton-results-secret.yaml': evalsymlink failure on '/source/developer/openshift/work/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml' : lstat /source/developer/openshift/work/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml: no such file or directory': must build at directory: not a valid directory: evalsymlink failure on '/source/developer/openshift/work/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml' : lstat /source/developer/openshift/work/credentials/manifests/compute/tekton-results/tekton-results-secret.yaml: no such file or directory
> ry-run=client -o yaml
> ```
> 
> It looks like a kustomization.yaml is referencing something it shouldn’t, or dev_setup.sh is not creating a file that is needed.

That's because the name was changed to `tekton-results-db-secret.yaml`  in this PR but the CI is still looking to main branch instead of the PR branch.
`use-current-branch` option is not working correctly for example, here in this case setup_work_dir [will always fetch main ](https://github.com/openshift-pipelines/pipeline-service/blob/a61967b3dc88ff2b22cdf1186b1a4740abc327fd/developer/openshift/dev_setup.sh#L272)

File: developer/openshift/gitops/argocd/pipeline-service-storage.yaml [hunk 1]
```diff
@@ -22,8 +22,6 @@ spec:
       selfHeal: true
     syncOptions:
       - CreateNamespace=true
-      # workaround to make working both minio: newer and old 4.5.4
```
Comment (AndrienkoAleksandr) [inline developer/openshift/gitops/argocd/pipeline-service-storage.yaml:25]: Why did you remove this? I don't think that you will pass ci test without this workaround

**Comment by Roming22** [comment]
@Mo3m3n I'm fine with your explanation. I'll merge the PR and we'll fail forward if necessary.

File: developer/openshift/gitops/argocd/pipeline-service-storage.yaml (hunk 1 above)
Comment (gabemontero) [reply developer/openshift/gitops/argocd/pipeline-service-storage.yaml:25]: @Mo3m3n did you address @AndrienkoAleksandr 's question elsewhere ?  If not, can you clarify why this was removed?

thanks

Comment (Mo3m3n) [reply developer/openshift/gitops/argocd/pipeline-service-storage.yaml:25]: Actually I have mistakenly removed this  line and I have not seen this comment, I will fix it  in a subsequent PR that aims get tests to green again in main

Comment (Mo3m3n) [reply developer/openshift/gitops/argocd/pipeline-service-storage.yaml:25]: @gabemontero @AndrienkoAleksandr can you provide more context about this ?
I am willing to fix this and provide a more detailed explanation
>  workaround to make working both minio: newer and old 4.5.4

Can we at least tell which minio part is going to fail and possibly why?

Comment (gabemontero) [reply developer/openshift/gitops/argocd/pipeline-service-storage.yaml:25]: I'll defer to @AndrienkoAleksandr to answer your question @Mo3m3n - he was the one who investigated / put in this workaround.

I had just caught the fact that this PR merged without resolving this thread.

**Pull request merged**